src/script/E2EIdentity/E2EIdentityVerification.ts

```diff
@@ -81,17 +89,28 @@ export async function getConversationVerificationState(groupId: string) {
 /**
  * Checks if E2EI has active certificate.
  */
-const fetchSelfDeviceIdentity = async (): Promise<WireIdentity | undefined> => {
-  const conversationState = container.resolve(ConversationState);
-  const selfMLSConversation = conversationState.getSelfMLSConversation();
-  const userIdentities = await getAllGroupUsersIdentities(selfMLSConversation.groupId);
+const getSelfDeviceIdentity = async (): Promise<WireIdentity | undefined> => {
+  let selfMLSConversationGroupId: string;
+  try {
+    // Try to get the self MLS conversation from the conversation state
+    const conversationState = container.resolve(ConversationState);
+    selfMLSConversationGroupId = conversationState.getSelfMLSConversation().groupId;
+  } catch {
+    // If the conversation state is not available, try to get the self MLS conversation from backend
+    const selfMLSConversation = await getCoreConversationService().getMLSSelfConversation();
+    selfMLSConversationGroupId = selfMLSConversation.group_id;
+  }
+
+  const userIdentities = await getAllGroupUsersIdentities(selfMLSConversationGroupId);
 
   if (!userIdentities) {
     return undefined;
   }
 
-  const currentClientId = selfMLSConversation.selfUser()?.localClient?.id;
-  const userId = selfMLSConversation.selfUser()?.qualifiedId;
+  const core = container.resolve(Core);
+
+  const currentClientId = core.clientId;
+  const userId = {id: core.userId, domain: core.backendFeatures.domain};
```

Comment:
We should probably change this to a fully qualified ID on core side :D

Reply:
Agreed!